integration_tests/test_ibc.py

```diff
@@ -261,3 +265,99 @@ def check_ica_tx():
 
     # check if the funds are reduced in interchain account
     assert cli_host.balance(ica_address, denom="basecro") == 50000000
+
+
+def test_cronos_transfer_source_tokens(ibc):
+    """
+    test sending crc20 tokens originated from cronos to crypto-org-chain
+    """
+    output = subprocess.getoutput(
+        f"curl -s -X GET 'http://127.0.0.1:{ibc.hermes.port}/state' | jq"
+    )
+    assert json.loads(output)["status"] == "success"
+
+    # deploy crc21 contract
+    w3 = ibc.cronos.w3
+    contract = deploy_contract(w3, CONTRACTS["TestERC21Source"])
+
+    # setup the contract mapping
+    cronos_cli = ibc.cronos.cosmos_cli()
+
+    print("crc21 contract", contract.address)
+    denom = f"cronos{contract.address}"
+
+    print("check the contract mapping not exists yet")
+    with pytest.raises(AssertionError):
+        cronos_cli.query_contract_by_denom(denom)
+
+    rsp = cronos_cli.update_token_mapping(
+        denom, contract.address, "DOG", 6, from_="validator"
```

Comment:
Can you add a negative case for an invalid denom?

Reply:
I think the token name and decimals could be fetched from the evm contract automatically.

Reply:
added a small negative case